src/util/resourceValidationUtils.js

```diff
@@ -0,0 +1,124 @@
+const { resolveSchema } = require('@projecttacoma/node-fhir-server-core');
+const axios = require('axios');
+const { v4: uuidv4 } = require('uuid');
+
+/**
+ * Checks body of request by querying a FHIR validation server. Updates the submitted resource to
+ * include validated profiles in the meta.profile field. Responds with 400 if body is not valid FHIR
+ * 500 if validation server error
+ * @param {Object} req request object of an express request
+ * @param {Object} res response object of an express request
+ * @param {Function} next express function which passes info along after clearing the middleware
+ */
+async function validateFhir(req, res, next) {
```

Comment:
A lot of this function is uncovered by unit tests. Any way we can try to bump the numbers up? Hopefully we can just mimick the inputs easily. If it's too much then no worries.